From 550a1a3ca935ffcfe458e6b5f588ce1e237d4404 Mon Sep 17 00:00:00 2001
From: Jason Rogena <j.rogena@cgiar.org>
Date: Wed, 21 May 2014 14:41:35 +0300
Subject: [PATCH 160/213] Bug Fixes Fixed two bugs    - SQLite database now
 being trunctated before updated farmer      data is inserted    - Size of the
 previous service spinner checked before trying      to get data out of it

---
 .../main/java/org/cgiar/ilri/mistro/farmer/AddEventActivity.java    | 4 ++--
 .../main/java/org/cgiar/ilri/mistro/farmer/backend/DataHandler.java | 6 ++++++
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/AddEventActivity.java b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/AddEventActivity.java
index 74a6dc5..b682ade 100644
--- a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/AddEventActivity.java
+++ b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/AddEventActivity.java
@@ -49,7 +49,7 @@ public class AddEventActivity extends SherlockActivity implements MistroActivity
 {
     private Boolean cacheData;
 
-    public static final String TAG="AddEventActivity";
+    public  final String TAG="AddEventActivity";
     private final String dateFormat="dd/MM/yyyy";
 
     public static final String KEY_MODE="mode";
@@ -679,7 +679,7 @@ public class AddEventActivity extends SherlockActivity implements MistroActivity
                     }
                     //jsonObject.put("bullEarTagNo", bullETNACTV.getText().toString());
                     jsonObject.put("noOfServicingDays", noOfServicingDaysET.getText().toString());
-                    if(servicingIDs != null) {
+                    if(servicingIDs != null && servicingS.getSelectedItemPosition() != -1) {
                         jsonObject.put("parentEvent", servicingIDs.get(servicingS.getSelectedItemPosition()));
                     }
                     if(selectedEvent.equals("Birth")){
diff --git a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/backend/DataHandler.java b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/backend/DataHandler.java
index 38e4b50..8e8e174 100644
--- a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/backend/DataHandler.java
+++ b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/backend/DataHandler.java
@@ -483,6 +483,12 @@ public class DataHandler
         //insert farmer data
         if(writableDB.isOpen()){
             try{
+                //remove all the data associated to farmer
+                databaseHelper.runTruncateQuery(writableDB, databaseHelper.TABLE_FARMER);
+                databaseHelper.runTruncateQuery(writableDB, databaseHelper.TABLE_COW);
+                databaseHelper.runTruncateQuery(writableDB, databaseHelper.TABLE_MILK_PRODUCTION);
+                databaseHelper.runTruncateQuery(writableDB, databaseHelper.TABLE_EVENT);
+
                 String[] columns = new String[]{"id","name","mobile_no","location_county","location_district","gps_longitude", "gps_latitude", "date_added", "sim_card_sn"};
                 String[] columnValues = new String[columns.length];
 
-- 
2.7.1

